### ISSUE TITLE
Fix compiler warnings

### ISSUE THREAD

**Issue by matthew-d-jones** [ess-dmsc/kafka-to-nexus#200]
Turning on `-pedantic -Wextra` would help us further improve code quality by picking up problems like unused parameters.

I tried adding the flags and relatively few errors are being thrown, most are warnings about the LOG macro:
```
ISO C++11 requires at least one argument for the "..." in a variadic macro
```
See https://twitter.com/arne_mertz/status/1008611706799575040 for possible workarounds.

**Comment by dominikwerder** [comment]
Got fixed by #198 and #201

**Comment by mattclarke** [comment]
Moved to https://jira.esss.lu.se/browse/DM-1164
